default.nix

```diff
@@ -357,4 +357,100 @@ in
           bw --help
           touch $out
         '';
+
+  # This tests and also serves as an example on how to handle bin
+  # dependencies.
+  #
+  # Typically bin dependencies are shipped with the npm package. These
+  # packages have a postinstall step that checks if the binary works and
+  # fallbacks to compiling from source otherwise. Because the binary isn't
+  # linked against NixOS paths the binary fails.
+  #
+  # So here we use the approach of skipping the postinstall, then using
+  # autopatchelf to fix all the binaries, and finally run the postinstall
+  # manually.
+  #
+  # NOTE: there are other phases such as "preinstall" and "install" that are
+  #       being ignored right now. In my opinion any npm package that depends
+  #       on those is broken.
+  #       See https://docs.npmjs.com/misc/scripts
+  bin-deps = buildPackage ./test/bin-deps {
+    nativeBuildInputs = with pkgs; [
+      autoPatchelfHook
+
+      nodejs.passthru.python
+
+      # These tend to be needed when
+      autoconf
+      automake
+      gettext
+      gnum4
+      gnused
+      nasm
+      pkgconfig
+    ];
+
+    postConfigure = ''
+      ${nodejs-headers-installer}
```

Comment:
I remember having to do something like this, and I think it worked without having to write anything to `HOME`. Though maybe it was something different?

```nix
''
...
export CPATH="${pkgs.nodejs-10_x}/include/node:$CPATH"
npm install --nodedir=${pkgs.nodejs-10_x}/include/node
...
''
```

Reply:
Thanks. I managed to avoid failing to download the headers with `npmCommands = ["npm install --nodedir=${pkgs.nodejs}/include/node --loglevel verbose"];`

(Although it eventually failed because there was no /usr/bin/env in a sandboxed build)